pageserver/src/restore_local_repo.rs

```diff
@@ -241,3 +242,133 @@ pub fn import_timeline_wal(walpath: &Path, timeline: &dyn Timeline, startpoint:
     info!("reached end of WAL at {}", last_lsn);
     Ok(())
 }
+
+///
+/// Helper function to parse a WAL record and call the Timeline's PUT functions for all the
+/// relations/pages that the record affects.
+///
+pub fn save_decoded_record(
+    timeline: &dyn Timeline,
+    decoded: DecodedWALRecord,
+    recdata: Bytes,
+    lsn: Lsn,
+) -> Result<()> {
+    // Figure out which blocks the record applies to, and "put" a separate copy
+    // of the record for each block.
+    for blk in decoded.blocks.iter() {
```

Comment:
`iter()` is mostly unnecessary; any type that implements `IntoIterator` can be used in a for loop.

In this case, to avoid trying to consume the inner `Vec` I would tend to use `for blk in &decoded.blocks { ... }`. I'm not sure whether that's "more idiomatic", though.

Reply:
I'd argue that `iter()` is a nice no-brainer, because it performs auto ref/deref. Which is to say that edits that change `Vec` to `&Vec` or `&&Vec` are not affected, whereas `for _ in &v` doesn't work when `v: &Vec` or something like that.

```rust
fn main() {
    let v = vec![1, 2, 2];
    let v_ref = &v;

    // :|
    for x in &v_ref {
        println!("{}", x);
    }

    // :D
    for x in (&v_ref).iter() {
        println!("{}", x);
    }
}
```

Reply:
This function should probably take a `&DecodedWALRecord` as argument, instead of pass-by-value `DecodedWALRecord`. I'll open a separate PR for that.